js/controller.js

```diff
@@ -553,6 +554,20 @@
                     $scope.focus = "timer";
                 }
             });
+
+            // Function to refresh TV show data
+            var refreshTVShows = function () {
+                console.log ("Refreshing TVShows");
+                $scope.tvshows = null;
+                TVShowService.refreshTVShows().then(function() {
+                  $scope.tvshows = TVShowService.getTVShows();
+                });
+            };
+
+            // Register refresh of TV shows once a day
+            if (typeof config.tvshows !== 'undefined'){
+                registerRefreshInterval(refreshTVShows, 24 * 60); // 24 hours * 60 min -- update once a day
```

Comment:
Existing services have a configurable refresh time (where applicable), could you add that to the config and use it here?